AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/Storage.smithy

```diff
@@ -192,8 +257,8 @@ operation GetKeyStorageInfo {
 
 @documentation(
 "Gets the ACTIVE branch key and the beacon key,
-and looks for a Mutation Lock,
-returning it if found.")
+and looks for a Mutation Commitment & Index,
```

Comment:
as a customer reading this, I have no clue what a Mutation Commitment is.

Reply:
Hang on... that is not fair... Mutation Commitment is a structure defined in the Storage Model.
Just like `HierarchicalSymmetric ` or `ActiveHierarchicalSymmetricVersion`. 

You can see that the JavaDocs for `MutationCommitment` populate it's Java class:
https://github.com/aws/aws-cryptographic-material-providers-library/blob/d4ba279972719c87f5e84f014f99bd565d8328a3/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/MutationCommitment.java

While it may not be explicitly defined right here,
in this paragraph,
it is defined in our documentation.

We also have to be careful.  
This is an extendable resource; we expect customers to implement this their selves.

It is not sufficient to say: 
"Gets the ACTIVE branch key, the beacon key, and any metadata on an in-flight Mutation",
as that is too generic.

Our Dafny implementation needs the well modeled 
Mutation Commitment 
and Mutation Index structures,
not just any conceptually equivalent.

But I can try and improve the customer experience here;
How does the following read to you:
```
Retrieves the items necessary to initialize a Mutation, 
while checking for any in-flight Mutations.  
These items are the ACTIVE branch key and the beacon key.
If a Mutation is already in-flight for this Branch Key,
the in-flight Mutation's Commitment and Index are also returned.
```